python/cuml/internals/api_context_managers.py

```diff
@@ -87,6 +108,19 @@ def set_api_output_type(output_type: str):
 
 
 def set_api_memory_type(mem_type):
+    """
+    Sets the memory type for the current internal API context.
+
+    Parameters
+    ----------
+    mem_type : str
+        The desired memory type.
+
+    Raises
+    ------
+    AssertionError
+        If the root context manager is not set.
```

Comment:
In general, we do not want developers to have to dig too deeply into the internals of this, especially worrying about what the "root context manager" is or why it is important. Instead, let's describe under what circumstances the root context manager might not be set.